packages/@tailwindcss-vite/src/index.ts

```diff
@@ -277,24 +277,30 @@ function optimizeCss(
   input: string,
   { file = 'input.css', minify = false }: { file?: string; minify?: boolean } = {},
 ) {
-  return transform({
-    filename: file,
-    code: Buffer.from(input),
-    minify,
-    sourceMap: false,
-    drafts: {
-      customMedia: true,
-    },
-    nonStandard: {
-      deepSelectorCombinator: true,
-    },
-    include: Features.Nesting,
-    exclude: Features.LogicalProperties,
-    targets: {
-      safari: (16 << 16) | (4 << 8),
-    },
-    errorRecovery: true,
-  }).code.toString()
+  function optimize(code: Buffer | Uint8Array) {
+    return transform({
+      filename: file,
+      code,
+      minify,
+      sourceMap: false,
+      drafts: {
+        customMedia: true,
+      },
+      nonStandard: {
+        deepSelectorCombinator: true,
+      },
+      include: Features.Nesting,
+      exclude: Features.LogicalProperties,
+      targets: {
+        safari: (16 << 16) | (4 << 8),
+      },
+      errorRecovery: true,
+    }).code
+  }
+
+  // Running Lightning CSS twice to ensure that adjacent rules are merged after
+  // nesting is applied. This creates a more optimized output.
+  return optimize(optimize(Buffer.from(input))).toString()
```

Comment:
I wonder if we can get away with running `optimize` just once if `lightningcss` is configured as the CSS processor... It might have a different config so probably not 😬

Reply:
Yeah that's tricky. Our config sets Safari to a version that doesn't understand nesting yet.

One suggestion that Devon had was to put `@media` queries on the outside instead of nested to do the merging, but the current change is a bit easier especially once Lightning CSS can handle this natively.